structopt-derive/src/lib.rs

```diff
@@ -84,6 +84,14 @@ fn gen_augmentation(fields: &Punctuated<Field, Comma>, app_var: &Ident) -> quote
         .filter_map(|field| {
             let attrs = Attrs::from_field(field);
             if attrs.is_subcommand() { return None; }
+
+            if attrs.should_flatten() {
+                let ty = &field.ty;
+                return Some(quote!{
+                    let #app_var = #ty ::augment_clap(#app_var);
```

Comment:
Note: this errors out with a somewhat strange message if `#ty` is something like `Option<Foo>`. Is there a good way to check for this?

Reply:
```rust
let #app_var = <#ty>::augment_clap(#app_var);
```

Maybe you can update other similar code gen.